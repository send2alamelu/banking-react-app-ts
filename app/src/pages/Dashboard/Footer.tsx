import { useHistory } from 'react-router-dom';

import BottomNavigationAction from '@mui/material/BottomNavigationAction';
import SendIcon from '@mui/icons-material/Send';

import BottomNav from '../../components/Common/BottomNav/BottomNav';

import { ROUTE_PATH } from '../../constants/RoutePath';
const { TRANSACTION } = ROUTE_PATH;

export default function Footer() {
  const history = useHistory();

  const goToTransferForm = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault();
    history.push(TRANSACTION);
  }

  return (
    <BottomNav>
      <BottomNavigationAction label="Make a transfer" onClick={goToTransferForm} icon={<SendIcon />} />
    </BottomNav>
  );
}
