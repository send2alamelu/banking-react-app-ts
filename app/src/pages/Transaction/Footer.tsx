import { useState } from 'react';
import { useHistory } from 'react-router-dom';

import BottomNavigationAction from '@mui/material/BottomNavigationAction';
import SendIcon from '@mui/icons-material/Send';
import CancelIcon from '@mui/icons-material/Cancel';

import BottomNav from '../../Components/Common/BottomNav/BottomNav';

export default function Footer() {
  const history = useHistory();

  const cancelTransaction = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault();
    // TODO: Call API
    history.push("/dashboard");
  }

  const sendTransaction = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault();
    // TODO: Call API
    history.push("/dashboard");
  }

  return (
    <BottomNav>
      <BottomNavigationAction label="Cancel" onClick={cancelTransaction} icon={<CancelIcon />} />
      <BottomNavigationAction label="Submit" onClick={sendTransaction} icon={<SendIcon />} />
    </BottomNav>
  );
}
