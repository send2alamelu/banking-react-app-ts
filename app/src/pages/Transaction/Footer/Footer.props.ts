export default interface FooterProps {
  onSubmit: Function;
}