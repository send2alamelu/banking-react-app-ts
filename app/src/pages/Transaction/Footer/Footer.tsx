import { useState } from 'react';
import { useHistory } from 'react-router-dom';

import BottomNavigationAction from '@mui/material/BottomNavigationAction';
import SendIcon from '@mui/icons-material/Send';
import CancelIcon from '@mui/icons-material/Cancel';

import BottomNav from '../../../components/Common/BottomNav/BottomNav';
import { ROUTE_PATH } from '../../../constants/RoutePath';
import FooterProps from './Footer.props';

const { DASHBOARD } = ROUTE_PATH;

export default function Footer(props: FooterProps) {
  const history = useHistory();

  const cancelTransaction = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault();
    // TODO: Call API
    history.push(DASHBOARD);
  }

  const sendTransaction = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault();
    // TODO: Call API
    // history.push(DASHBOARD);
    props.onSubmit();
  }

  return (
    <BottomNav>
      <BottomNavigationAction label="Cancel" onClick={cancelTransaction} icon={<CancelIcon />} />
      <BottomNavigationAction label="Submit" onClick={sendTransaction} icon={<SendIcon />} />
    </BottomNav>
  );
}
