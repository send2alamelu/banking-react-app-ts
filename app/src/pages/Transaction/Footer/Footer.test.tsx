import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Footer from './Footer';

describe('Footer Component', () => {
  afterEach(() => {
    cleanup();
  });

  test('renders Footer with children', () => {
    const { asFragment } = render(<Footer onSubmit={jest.fn} />);
    expect(asFragment()).toMatchSnapshot();
  });
});