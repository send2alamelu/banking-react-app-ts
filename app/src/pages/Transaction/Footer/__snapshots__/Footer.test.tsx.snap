// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders Footer with children 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-xuhmcu"
  >
    <div
      class="MuiBottomNavigation-root css-16lloyr-MuiBottomNavigation-root"
    >
      <button
        class="MuiButtonBase-root MuiBottomNavigationAction-root css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CancelIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
        >
          Cancel
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiBottomNavigationAction-root css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SendIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
        >
          Submit
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
